crates/core/src/sql/value/merge.rs

```diff
@@ -10,8 +10,18 @@ impl Value {
 			});
 		}
 		// Otherwise loop through every object field
-		for k in val.every(None, false, false).iter() {
-			match val.pick(k) {
+		for k in val.every(None, true, false).iter() {
+			// Because we iterate every step, we need to this check
+			// If old & new are both objects, we do not want to completely
+			// replace the old object with the new object as that will drop
+			// all the fields in the old object that are not in the new object
+			let old = self.pick(k);
+			let new = val.pick(k);
+			if old.is_object() && new.is_object() {
+				continue;
```

Comment:
If they are both objects, shouldn't we merge them?

Reply:
That's what it will do 😄 
It will continue to iterate over the paths, where the next set of paths will be the object's properties. The test proofs it 😁

Reply:
The test ends with

```hcl
{
    test: true,
    name: {
	    title: 'Mr',
	},
}
```

shouldn't it be this instead?

```hcl
{
    test: true,
    name: {
	    title: 'Mr',,
        initials: NONE,
	},
}
```

Is `initials` dropped because it has a value of `NONE`? It would be nice to do another merge when `name` is already an object to show clearly how this works. I would set `title` to something else and and add another property to `name`.

Reply:
Sure